derive/src/pyclass.rs

```diff
@@ -1,91 +1,138 @@
 use super::rustpython_path_attr;
-use proc_macro2::{Span, TokenStream as TokenStream2};
+use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
+use std::collections::HashMap;
 use syn::{Attribute, AttributeArgs, Ident, ImplItem, Item, Lit, Meta, MethodSig, NestedMeta};
 
-enum MethodKind {
+#[derive(PartialEq, Clone, Copy)]
+enum ClassItemKind {
     Method,
-    Property,
+    PropertyGetter,
+    PropertySetter,
 }
 
-impl MethodKind {
-    fn to_ctx_constructor_fn(&self) -> Ident {
-        let f = match self {
-            MethodKind::Method => "new_rustfunc",
-            MethodKind::Property => "new_property",
-        };
-        Ident::new(f, Span::call_site())
-    }
+struct ClassItem {
```

Comment:
In the spirit of "make invalid states unrepresentable", could this and `ClassKindKind` be combined into an enum like:

```rustt
enum ClassItem {
    Method {
        name: String,
        ...
    },
    Property {
        setter: bool
    }
```

i.e. each variant represents an attribute, each field represents metadata and/or an "argument" to the attribute

Reply:
That feels better, thanks.